subsys/settings/settings.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2018 Laczen
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2015 Runtime Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <string.h>
+
+#include <settings/settings.h>
+#include "settings_priv.h"
+
+sys_slist_t settings_handlers;
+
+static u8_t settings_cmd_inited;
+
+void settings_store_init(void);
+
+void settings_init(void)
+{
+	if (!settings_cmd_inited) {
+		sys_slist_init(&settings_handlers);
+		settings_store_init();
+
+		settings_cmd_inited = 1U;
+	}
+}
+
+int settings_register(struct settings_handler *handler)
+{
```

Comment:
I have seen the commit that changed this, however I don't see a reason why the same name could not be used several times. It can be advantageous to register several handlers with the same name, each of the set routines would be called and could provide handling of different items under this same name.

Reply:
You are correct, it will only call the first settings_handler that is registered with this name. I will add the check.